app/models/operatingsystem.rb

```diff
@@ -46,6 +46,13 @@ class Operatingsystem < ApplicationRecord
 
   default_scope -> { order(:title) }
 
+  # Find all operatingsystems that are duplicates of the given operating system according to all unique constraints
+  def self.duplicate_of(os)
+    scope = where(name: os.name, major: os.major, minor: os.minor)
+    scope = scope.or(where(description: os.description)) if os.description.present?
+    scope.or(where(title: os.title || os.generate_title))
```

Comment:
Looks like in all examples this returns an array. I would then expect the method name to reflect this returns a plural (e.g .`duplicates_of`).